sdk/core/core/CMakeLists.txt

```diff
@@ -45,6 +45,8 @@ add_library (
 
 if(MSVC)
   target_compile_options(az_core PRIVATE /Wall /WX /wd5031 /wd4668 /wd4820 /wd4255 /analyze)
+elseif(CMAKE_C_COMPILER_ID MATCHES "Clang")
+  target_compile_options(az_core PRIVATE -Wall -Wextra -pedantic -Werror -Wdocumentation -Wdocumentation-unknown-command -Wcast-qual)
```

Comment:
This causes a problem with clang-cl. In that case CMAKE_C_COMPILER_ID will be "Clang", however -Wall will be translated by clang-cl as /Wall, which is equivalent to -Weverything in clang (sigh).

The solution is to instead say -Xclang -Wall .....